### PR TITLE
refactor: remove bluebird

### PR DESCRIPTION
**Pull request by SimonSchick**
Currently playing around with removing bluebird.

Breaking changes are:
1) No more bluebird (duh)
2) Using own TimeoutError.

If possible I'd like to get this released as a major RC.

### CODE REVIEW

**Comment by DiegoRBaquero** [comment]
Honest question, why was bluebird used when moving from `node-pool`?

I feel like Bluebird could be removed from `sequelize` as a whole, but maybe there's a reason as to why it is there.

**Review by eseliger — APPROVED** [review]

**Comment by SimonSchick** [comment]
@DiegoRBaquero I have no idea, probably consistency/ease of use.
I'm playing around with purging bluebird from sequelize but that's a lot of work.

**Comment by SimonSchick** [comment]
@DiegoRBaquero actually, sequelize has been around longer than the native promise implementations were near stable enough to use.
With modern async/await it's fair to consider it's removal.

**Comment by DiegoRBaquero** [comment]
Yes, I'm coming from the basis that sequelize v5 uses Node 6+, which is where `node-pool` was removed. So we can consider removing it and use the native implementation.

File: lib/Pool.js [hunk 1]
```diff
@@ -453,9 +453,9 @@ class Pool {
     this._removeIdleScheduled = false;
     clearTimeout(this._removeIdleTimer);
 
-    return Promise.fromCallback(callback => {
+    return new Promise(res => {
```
Comment (sushantdhiman) [inline lib/Pool.js]: `resolve`

Comment (SimonSchick) [reply lib/Pool.js]: Oh, come on 😛

**Comment by SimonSchick** [comment]
@sushantdhiman could you please perform a major RC release?
Might also move this to the sequelize org :)

**Pull request merged**